src/lib.rs

```diff
@@ -36,6 +36,96 @@
 //! This crate provides utilities that make writing both clients and servers easier. You'll find
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
+//!
+//! # Example
+//! ```rust
+//! # use std::pin::Pin;
+//! # use std::boxed::Box;
+//! # use tokio::sync::mpsc;
+//! # use tokio::io::{AsyncWrite, AsyncRead};
+//! # use futures_core::task::{Context, Poll};
+//! # use futures_util::{never::Never, future::{poll_fn, ready, Ready}};
+//! # use tokio_tower::pipeline;
+//! # use core::fmt::Debug;
+//! type StdError = Box<dyn std::error::Error + Send + Sync + 'static>;
+//!
+//! /// Wrapper around our mpsc channel transport.
+//! ///
+//! /// mpsc::Sender and mpsc::Receiver operate at the level of individual items,
+//! /// but tokio-tower requires Sink + Stream. To bridge this gap,
+//! /// we have to provide implementations of Sink and Stream.
```

Comment:
I think what is missing here is an explanation of _why_ `Sink + Stream` is required. I would try to write it out since this is a top-level example. Something like:

> At its core, tokio-tower wraps a type that is `Sink + Stream`. On the client side, the `Sink` is used to send requests, and the `Stream` is used to receive responses (from the server) to those requests. On the server side, the `Stream` is used to receive requests, and the `Sink` is used to send the responses. If we want to use `mpsc` to send requests and responses between a client and server, we therefore need _two_ channels, one that lets requests flow from the client to the server, and one that lets responses flow the other way. In this example, requests and responses are both of type `T`, but the two types are usually different in anything that is not just an echo server like in this example.

Reply:
The first half of that explanation is essentially a clearer wording for the text currently in the "Transports" section, so it could replace it. 

The second half (starting with "If we want to use `mpsc`") could follow the text that is currently there?

Reply:
Works for me!